devices/uart-imxrt106x/uart.c

```diff
@@ -511,15 +511,14 @@ static ssize_t uart_write(unsigned int minor, const u8 *buff, unsigned int len)
 }
 
 
-static ssize_t uart_safeWrite(unsigned int minor, addr_t offs, const u8 *buff, unsigned int len)
+static ssize_t uart_safeWrite(unsigned int minor, addr_t offs, const void *buff, size_t len)
 {
-	unsigned int l;
+	ssize_t res;
+	size_t l;
 
-	for (l = 0; len;) {
-		if ((l = uart_write(minor, buff, len)) < 0)
+	for (l = 0; l < len; l += res) {
+		if ((res = uart_write(minor, (const char *)buff + l, len - l)) < 0)
```

Comment:
Comment for the future - it's best not to use lowercase L (l) because it can be mistaken either with 1 or I (uppercase i)

Reply:
l1I - which is which? 😄